File: e2e-test/rfc-test/web/src/ExplorerInput.ts
/**
 * ExplorerInput element implementation
 * 
 * This is a complete, self-contained element class that you can modify directly.
 * The class extends HTMLElement and can be discovered by the build system.
 * 
 * TODO: Customize this class by:
 * 1. Implementing property setters to update your element
 * 2. Adding event emission calls where needed
 * 3. Implementing lifecycle methods (connectedCallback, disconnectedCallback)
 * 4. Optionally using Shadow DOM for encapsulation
 */
export class ExplorerInput extends HTMLElement {
  
  constructor() {
    super();
    // TODO: Initialize your custom element
    // Example: this.attachShadow({ mode: 'open' });
    // Example: this.innerHTML = '<div>Hello World</div>';
  }

  connectedCallback() {
    // TODO: Element added to DOM
    // This is called when the element is inserted into the DOM
  }

  disconnectedCallback() {
    // TODO: Element removed from DOM
    // This is called when the element is removed from the DOM
  }

  attributeChangedCallback(name: string, oldValue: string, newValue: string) {
    // TODO: Handle attribute changes
    // This is called when observed attributes change
  }

  static get observedAttributes() {
    // TODO: Return array of attribute names to observe
    // Example: return ['value', 'placeholder'];
    return [];
  }

  setBindinput(bindinput?: (e: Event) => void): void {
    // TODO: Update your element with bindinput
    // Access the element with: this.someProperty = bindinput
    // Example implementation:
    // this.setAttribute('bindinput', String(bindinput));
  }

  setClassName(className?: string): void {
    // TODO: Update your element with className
    // Access the element with: this.someProperty = className
    // Example implementation:
    // this.setAttribute('className', String(className));
  }

  setId(id?: string): void {
    // TODO: Update your element with id
    // Access the element with: this.someProperty = id
    // Example implementation:
    // this.setAttribute('id', String(id));
  }

  setStyle(style?: string | CSSStyleDeclaration): void {
    // TODO: Update your element with style
    // Access the element with: this.someProperty = style
    // Example implementation:
    // this.setAttribute('style', String(style));
  }

  setValue(value?: string | undefined): void {
    // TODO: Update your element with value
    // Access the element with: this.someProperty = value
    // Example implementation:
    // this.setAttribute('value', String(value));
  }

  setMaxlines(maxlines?: number): void {
    // TODO: Update your element with maxlines
    // Access the element with: this.someProperty = maxlines
    // Example implementation:
    // this.setAttribute('maxlines', String(maxlines));
  }

  setPlaceholder(placeholder?: string): void {
    // TODO: Update your element with placeholder
    // Access the element with: this.someProperty = placeholder
    // Example implementation:
    // this.setAttribute('placeholder', String(placeholder));
  }

  // Helper method for event emission
  // Call this method to send events back to the JavaScript layer
  protected emitEvent(name: string, value?: Record<string, any>): void {
    const event = new CustomEvent(name, {
      detail: value,
      bubbles: true,
      cancelable: true
    });
    this.dispatchEvent(event);
  }

  // Example event emission methods (uncomment and modify as needed):
  /*
  private emitClick() {
    this.emitEvent("click", { timestamp: Date.now() });
  }

  private emitChange(newValue: string) {
    this.emitEvent("change", { value: newValue });
  }
  */
}

// Register the custom element
customElements.define('explorerinput', ExplorerInput);

File: e2e-test/module-element-testing/ios/src/ExplorerInput.swift
import Foundation
import UIKit

/**
 * ExplorerInput element implementation
 * 
 * This is a complete, self-contained element class that you can modify directly.
 * The class extends LynxUI<UIView> and can be discovered by the build system.
 * 
 * TODO: Customize this class by:
 * 1. Implementing the createView() method to return your desired view type
 * 2. Implementing property setters to update your view
 * 3. Adding event emission calls where needed
 * 4. Optionally changing the UIView type to a more specific iOS view
 */
@objcMembers
public final class ExplorerInput: LynxUI<UIView> {

    public override func createView() -> UIView {
        // TODO: Create and configure your UIView
        // You can change UIView to any iOS view type (UIButton, UITextField, etc.)
        let view = UIView()
        
        // TODO: Configure your view properties here
        // Example: view.backgroundColor = UIColor.white
        // Example: view.layer.cornerRadius = 8
        
        return view
    }

    @objc func setBindinput(_ bindinput: (Any) -> Void?) {
        // TODO: Update your view with bindinput
        // Access the native view with: view.bindinput = bindinput
        // Example implementation:
        // view.bindinput = bindinput
    }

    @objc func setClassName(_ className: String?) {
        // TODO: Update your view with className
        // Access the native view with: view.className = className
        // Example implementation:
        // view.className = className
    }

    @objc func setId(_ id: String?) {
        // TODO: Update your view with id
        // Access the native view with: view.id = id
        // Example implementation:
        // view.id = id
    }

    @objc func setStyle(_ style: Any?) {
        // TODO: Update your view with style
        // Access the native view with: view.style = style
        // Example implementation:
        // view.style = style
    }

    @objc func setValue(_ value: Any?) {
        // TODO: Update your view with value
        // Access the native view with: view.value = value
        // Example implementation:
        // view.value = value
    }

    @objc func setMaxlines(_ maxlines: Double?) {
        // TODO: Update your view with maxlines
        // Access the native view with: view.maxlines = maxlines
        // Example implementation:
        // view.maxlines = maxlines
    }

    @objc func setPlaceholder(_ placeholder: String?) {
        // TODO: Update your view with placeholder
        // Access the native view with: view.placeholder = placeholder
        // Example implementation:
        // view.placeholder = placeholder
    }

    // Helper method for event emission
    // Call this method to send events back to the JavaScript layer
    protected func emitEvent(name: String, value: [String: Any]? = nil) {
        let detail = LynxCustomEvent(sign: getSign(), name: name)
        if let value = value {
            for (key, v) in value {
                detail.addDetail(key, value: v)
            }
        }
        getLynxContext().getEventEmitter().sendCustomEvent(detail)
    }

    // Example event emission methods (uncomment and modify as needed):
    /*
    private func emitClick() {
        emitEvent(name: "click", value: ["timestamp": Date().timeIntervalSince1970])
    }

    private func emitChange(newValue: String) {
        emitEvent(name: "change", value: ["value": newValue])
    }
    */
}